isic_challenge_scoring/classification.py

```diff
@@ -61,9 +69,36 @@ def __init__(
         )
 
         self.overall = self.aggregate.at['balanced_accuracy']
-        self.validation = metrics.balanced_multiclass_accuracy(
-            truth_probabilities, prediction_probabilities, truth_weights.validation_weight
-        )
+
+        if validation_metric:
+            if validation_metric == ValidationMetric.BALANCED_ACCURACY:
+                self.validation = metrics.balanced_multiclass_accuracy(
+                    truth_probabilities, prediction_probabilities, truth_weights.validation_weight
+                )
+            elif validation_metric == ValidationMetric.AVERAGE_PRECISION:
+                per_category_ap = pd.DataFrame(
+                    [
+                        metrics.average_precision(
+                            truth_probabilities[category],
+                            prediction_probabilities[category],
+                            truth_weights.validation_weight,
+                        )
+                        for category in categories
+                    ]
+                )
+                self.validation = per_category_ap.mean(axis='index')
```

Comment:
Assuming `per_category_ap` ends up as a `Series`, I don't think the additional `axis=` specification should be necessary.